src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/step/rollup/AttemptCreateRollupJobStep.kt

```diff
@@ -0,0 +1,123 @@
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexmanagement.indexstatemanagement.step.rollup
+
+import com.amazon.opendistroforelasticsearch.indexmanagement.indexstatemanagement.model.ManagedIndexMetaData
+import com.amazon.opendistroforelasticsearch.indexmanagement.indexstatemanagement.model.managedindexmetadata.ActionProperties
+import com.amazon.opendistroforelasticsearch.indexmanagement.indexstatemanagement.model.managedindexmetadata.StepMetaData
+import com.amazon.opendistroforelasticsearch.indexmanagement.indexstatemanagement.step.Step
+import com.amazon.opendistroforelasticsearch.indexmanagement.rollup.action.index.IndexRollupAction
+import com.amazon.opendistroforelasticsearch.indexmanagement.rollup.action.index.IndexRollupRequest
+import com.amazon.opendistroforelasticsearch.indexmanagement.rollup.action.start.StartRollupAction
+import com.amazon.opendistroforelasticsearch.indexmanagement.rollup.action.start.StartRollupRequest
+import com.amazon.opendistroforelasticsearch.indexmanagement.rollup.model.ISMRollup
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.apache.logging.log4j.LogManager
+import org.elasticsearch.action.support.WriteRequest
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.service.ClusterService
+import org.elasticsearch.index.engine.VersionConflictEngineException
+import java.lang.Exception
+
+class AttemptCreateRollupJobStep(
+    val clusterService: ClusterService,
+    val client: Client,
+    val ismRollup: ISMRollup,
+    managedIndexMetaData: ManagedIndexMetaData
+) : Step(name, managedIndexMetaData) {
+
+    private val logger = LogManager.getLogger(javaClass)
+    private var stepStatus = StepStatus.STARTING
+    private var info: Map<String, Any>? = null
+    private var rollupId: String? = null
+    private var previousRunRollupId: String? = null
+    private var hasPreviousRollupAttemptFailed: Boolean? = null
+
+    override fun isIdempotent() = true
+
+    override suspend fun execute(): Step {
+        previousRunRollupId = managedIndexMetaData.actionMetaData?.actionProperties?.rollupId
+        hasPreviousRollupAttemptFailed = managedIndexMetaData.actionMetaData?.actionProperties?.hasRollupFailed
+
+        // Creating a rollup job
+        val rollup = ismRollup.toRollup(indexName)
+        rollupId = rollup.id
+        logger.info("Attempting to create a rollup job $rollupId for index $indexName")
+
+        val indexRollupRequest = IndexRollupRequest(rollup, WriteRequest.RefreshPolicy.IMMEDIATE)
+
+        try {
+            val response = withContext(Dispatchers.IO) { client.execute(IndexRollupAction.INSTANCE, indexRollupRequest).actionGet() }
```

Comment:
As discussed offline let's change this to use the suspendUntil helper method we have instead of actionGet

Reply:
updated